src/TwigExtension.php

```diff
@@ -22,4 +24,94 @@ public function is_instanceof($value, string $type): bool
       || (\function_exists($func = 'is_'.$type) && $func($value))
       || $value instanceof $type;
   }
-}
\ No newline at end of file
+
+  /**
+   * Returns entity ids of entities with matching title/name/label.
+   *
+   * Supported entity types are node, taxonomy_term, group, and user.
+   *
+   * @param  string  $label
+   *   The entity label that we're looking for
+   * @param  string  $entity_type
+   *   The entity type.
+   * @param  string  $bundle_identifier
+   *   The entity bundle (may be empty)
+   * @param  bool  $check_access
+   *   Whether to check for entity access permission (default TRUE).
+   * @param  int  $limit
+   *   Restrict to number of results. Capped at no more than 100.
+   * @param  null  $langcode
+   *   (optional) For which language the entity should be rendered, defaults to
+   *   the current content language.
+   *
+   * @return null|array
+   *   An array of render arrays for the entities found, or NULL if the entity does not exist.
+   */
+  public function sbf_entity_ids_by_label(string $label, string $entity_type, string $bundle_identifier = '', $check_access = TRUE, $limit = 1, $langcode = NULL): ?array {
+    $label = \Drupal::database()->escapeLike($label);
+    $limit = min($limit, 100);
+    /** @var \Drupal\Core\Entity\Query\QueryInterface $query */
+    try {
+      switch($entity_type) {
```

Comment:
This is more a suggestion in terms of reducing code (not against repeating code when it adds to legibility, so take/leave)

You allow a set of Entities here, 'node', etc. Maybe checking with an in_array() first to return immediately if not present.
Then the $query = \Drupal::entityTypeManager()->getStorage($valid_type)->getQuery(); would be a single call.
And these are also common

```
->accessCheck(TRUE) // hope so!
->range(0,$limit); 
```

independently of the type
```
And same for  $ids = $query->execute();
```

your only, per type particularities are which field contains the "title" and "how" the $bundle_identifier needs to be processed. That can stay in your switch for sure.

That would reduce a lot of code

Reply:
I totally rewrote this section along the lines you suggested.

Reply:
Thanks so much. Good coding